src/passes/OptimizeInstructions.cpp

```diff
@@ -1369,7 +1372,8 @@ struct OptimizeInstructions
     }
     {
       double value;
-      if (matches(curr, binary(Abstract::Sub, any(), fval(&value))) &&
+      if (fastMath &&
+          matches(curr, binary(Abstract::Sub, any(), fval(&value))) &&
           value == 0.0) {
         // x - (-0.0)   ==>   x + 0.0
```

Comment:
What's wrong about this optimization without `fastMath`? Is it that it doesn't change non-canonical NaN bits but it should?

Reply:
Hmm, actually, re-reading it now I'm not sure. It doesn't remove a math operation (unlike the others), so it would still change NaNs as expected, I think? I guess we'd need to read the spec (wasm? IEEE?) carefully. If no one knows offhand, the safe thing may be to land this with a TODO for later.

Reply:
Imteresting article:
https://754r.ucbtest.org/background/nan-propagation.pdf
See "What should happen when two payloads are combined?"

Reply:
And yes seems to be IEEE754 doesn't specify how two NaNs with different payloads should be combined. I guess it should be reflect on wasm spec tests

Reply:
Btw webassembly spec specify Na N-propogation rules for fneg, fabs and fcopysign:
https://webassembly.github.io/spec/core/bikeshed/index.html#nan-propagation%E2%91%A0